clients/src/main/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManager.java

```diff
@@ -550,6 +550,11 @@ public ConsumerGroupHeartbeatRequestData buildRequestData() {
             } else {
                 // SubscribedTopicRegex - only sent if has changed since the last heartbeat
                 //                      - not supported yet
+                TreeSet<String> subscribedTopicNames = new TreeSet<>(this.subscriptions.subscription());
+                if (sendAllFields || !subscribedTopicNames.equals(sentFields.subscribedTopicNames)) {
+                    data.setSubscribedTopicNames(new ArrayList<>(this.subscriptions.subscription()));
+                    sentFields.subscribedTopicNames = subscribedTopicNames;
+                }
```

Comment:
With this addition we end up with the exact same code repeated for the `if` and `else`, so I would say we should find a better way of doing this. First solution that comes to mind is to remove the if/else. In the end, we have a single case to handle here: send explicit subscriptions (topic names) to the broker (from the HB Mgr POV and to the broker, it does not make a diff if the topic list came from a call to subscribe with topics or a call to subscribe with Pattern that we internally resolved on the client) 

When we take on the next task of supporting the new regex, we'll actually have to send something different here, so we can decide then how to best differentiate the 2 cases. For now, including this PR, we only support 1 case regarding the content of what we send in the HB regarding subscription. Makes sense?

Reply:
Thanks for catching this, @lianetm. I agree that if the logic is identical, there's no need for duplication. Any speculation as to the reason the `if` statement existed in the first place?

Reply:
I guess the if/else was initially added to set the structure that there will be 2 paths:
1- send the broker the explicit topic names to subscribe to
2- send the broker a regex, for the broker to resolve it
So, key point is that, with the concepts above, up to this PR (included), we're still only supporting path 1, so that' why I would say we better get rid of the if/else, and include it only when we do implement the 2nd path, and we properly think about the condition to use, and add a body to the else